bindings/python/unicorn/unicorn.py

```diff
@@ -958,95 +1134,63 @@ def context(self):
-        self._size = state[1]
-        self._context = ctypes.cast(ctypes.create_string_buffer(state[0], self._size), uc_context)
-        # __init__ won'e be invoked, so we are safe to set it here.
+    def __setstate__(self, state) -> None:
+        context, size, arch, mode = state
+
+        self._context = ctypes.cast(ctypes.create_string_buffer(context, size), uc_context)
+        self._size = size
+        self._arch = arch
+        self._mode = mode
+
+        # __init__ won't be invoked, so we are safe to set it here.
         self._to_free = False
-        self._arch = state[2]
-        self._mode = state[3]
 
-    def __bytes__(self):
+
+    def __bytes__(self) -> bytes:
         return ctypes.string_at(self.context, self.size)
 
-    def __del__(self):
+
+    def __del__(self) -> None:
         # We need this property since we shouldn't free it if the object is constructed from pickled bytes.
         if self._to_free:
-            _uc.uc_context_free(self._context)
-
-UC_HOOK_CODE_TYPE = Callable[[Uc, int, int, Any], None]
-UC_HOOK_INSN_INVALID_TYPE = Callable[[Uc, Any], bool]
-UC_HOOK_MEM_INVALID_TYPE = Callable[[Uc, int, int, int, int, Any], bool]
-UC_HOOK_MEM_ACCESS_TYPE = Callable[[Uc, int, int, int, int, Any], None]
-UC_HOOK_INTR_TYPE = Callable[[Uc, int, Any], None]
-UC_HOOK_INSN_IN_TYPE = Callable[[Uc, int, int, Any], int]
-UC_HOOK_INSN_OUT_TYPE = Callable[[Uc, int, int, int, Any], None]
-UC_HOOK_INSN_SYSCALL_TYPE = Callable[[Uc, Any], None]
-UC_HOOK_INSN_SYS_TYPE = Callable[[Uc, int, Tuple[int, int, int, int, int, int], Any], int]
-UC_HOOK_INSN_CPUID_TYPE = Callable[[Uc, Any], int]
+            uclib.uc_context_free(self._context)
+
+
 UC_MMIO_READ_TYPE = Callable[[Uc, int, int, Any], int]
 UC_MMIO_WRITE_TYPE = Callable[[Uc, int, int, int, Any], None]
-UC_HOOK_EDGE_GEN_TYPE = Callable[[Uc, uc_tb, uc_tb, Any], None]
-UC_HOOK_TCG_OPCODE_TYPE = Callable[[Uc, int, int, int, Any], None]
-
-UC_HOOK_CALLBACK_TYPE = Union[
-    UC_HOOK_CODE_TYPE, 
-    UC_HOOK_INSN_INVALID_TYPE, 
-    UC_HOOK_MEM_INVALID_TYPE, 
-    UC_HOOK_MEM_ACCESS_TYPE, 
-    UC_HOOK_INSN_IN_TYPE, 
-    UC_HOOK_INSN_OUT_TYPE,
-    UC_HOOK_INSN_SYSCALL_TYPE,
-    UC_HOOK_INSN_SYS_TYPE,
-    UC_HOOK_INSN_CPUID_TYPE,
-    UC_HOOK_EDGE_GEN_TYPE,
-    UC_HOOK_TCG_OPCODE_TYPE
-]
-
-# print out debugging info
-def debug():
-    archs = {
-        "arm": uc.UC_ARCH_ARM,
-        "arm64": uc.UC_ARCH_ARM64,
-        "mips": uc.UC_ARCH_MIPS,
-        "sparc": uc.UC_ARCH_SPARC,
-        "m68k": uc.UC_ARCH_M68K,
-        "x86": uc.UC_ARCH_X86,
-        "riscv": uc.UC_ARCH_RISCV,
-        "ppc": uc.UC_ARCH_PPC,
-    }
-
-    all_archs = ""
-    keys = archs.keys()
-    for k in sorted(keys):
-        if uc_arch_supported(archs[k]):
-            all_archs += "-%s" % k
-
-    major, minor, _combined = uc_version()
-
-    return "python-%s-c%u.%u-b%u.%u" % (
-        all_archs, major, minor, uc.UC_API_MAJOR, uc.UC_API_MINOR
-    )
+
+
+__all__ = ['Uc', 'UcContext']
```

Comment:
This might be a bit aggressive as stated above.

Reply:
Importing `*` from a library is considered a bad practice and should be avoided.
As far as I can tell the Unicorn bindings library is organized in way that `__init__.py` already imports all the needed modules and "flatten" the import path (i.e. you can import everything directly from `unicorn`; if you can't import - it means you shouldn't).

Keep in mind that the previous implementation kept everything in one big file, so there was no consideration of a clean namespace. Having said that, I think that moving to a new major version (uc2) is a good oportunity to make such changes that could break some scripts - but at the same time will be trivial to fix by the users by adding a few more imports.